Week8AssignmentNew/ConsoleApp5/HomePage1.cs

```diff
@@ -0,0 +1,260 @@
+                return 1;
+            }
+        }
+
+        public int DiplayTotalItemsPresent()
+        {
+            String TotalItems = driver.FindElement(this.totalItems).Text;
+            String[] Totalitems = TotalItems.Split(' ');
+            int NumberOfItemsOnPage = Int16.Parse(Totalitems[3]);
+            return (NumberOfItemsOnPage);
+        }
+        public bool ItemUnavailable(int index)
+        {
+            try
+            {
+                return driver.FindElement(By.XPath(string.Format(unavailableItem, index))).Displayed;
+            }
+            catch (Exception e)
+            {
+                return false;
+            }
+        }
+        internal void AddToCart()
+        {
+            int pageCounter = 1;
+            int ppageCount = GetNumberOfPages();            
+            WebDriverWait wait = new WebDriverWait(driver, TimeSpan.FromSeconds(30));
+            java.lang.Thread.sleep(1000);
+            Cart1 cart = new Cart1(driver);
+            while (pageCounter <= ppageCount)
+            {
+                int itemCount = driver.FindElements(By.XPath("(//img[contains(@alt,'Apple')])")).Count;
+                //int iitemsCount = DiplayTotalItemsPresent();
+                for (int index = 1; index <= itemCount; index++)
+                {
+                    bool itemUnavailable = ItemUnavailable(index);
+                    //bool itemAvailable = driver.FindElement(By.XPath(string.Format(UnavailableItem, index))).Displaye
+                    if (!itemUnavailable) // if item available
+                    {
+                        //add to cart.
+                        java.lang.Thread.sleep(1000);
+                        try
+                        {
+                            driver.FindElement(By.XPath(string.Format(availableitem, index))).Click();
+                        }
+                        catch (Exception e)
+                        {
+                            driver.Close();
+                        }
+                        java.lang.Thread.sleep(2000);
+
+                        //Window Handles implementation
+                        ArrayList tabs = new ArrayList(driver.WindowHandles);
+                        driver.SwitchTo().Window((string)tabs[1]);
+                        cart.AddItemToCart();
+                        java.lang.Thread.sleep(1000);
+                        driver.SwitchTo().Window(driver.WindowHandles.First());
+                    }
+                }
+                pageCounter++;
+                if (pageCounter <= ppageCount)
+                {
+                    ClickNextButton();
+                    Thread.Sleep(2000);
+                }                
+            }
+        }
+        public string FinalSearchResult()
+        {
+            string r = driver.FindElement(searchResult).Text;
+            return r;
+        }
+
+        public void ClickNextButton()
+        {
+            driver.FindElement(nextButton).Click();
```

Comment:
this approach makes sense when you are using a base class. In this case since you are directly accessing driver in page object, this line can be used directly where it needs to be used.